src/client/mesh.cpp

```diff
@@ -337,6 +337,8 @@ bool checkMeshNormals(scene::IMesh *mesh)
 
 	for (u32 i = 0; i < buffer_count; i++) {
 		scene::IMeshBuffer *buffer = mesh->getMeshBuffer(i);
+		if (!buffer->getVertexCount())
```

Comment:
This should probably at least get a comment on why it is done.

I also suggest to add a unit test (using the file from @appgurueu maybe?) to prevent regressions.

Reply:
Here it is: [bad_apple.b3d.zip](https://github.com/minetest/minetest/files/12706124/bad_apple.zip). The empty TRIS chunk causes an empty vertex buffer which causes a segfault without this check. Perhaps this should be fixed in the reader(s) instead (by just not adding empty mesh buffers at all, upholding the invariant that all mesh buffers must be nonempty) but for a quick fix this is definitely appropriate.

Reply:
Just look some lines below, where it accesses vertex 0. :P

Reply:
In LANGSEC terms, @appgurueu is probably right – a fix (and test) should be in the B3D loader.

(This does not mean this thing can not be done, it just means that the B3D loader should be fixed.)

Reply:
We don't have an easy means to test the entire model loading and using process "offline" and testing only `checkMeshNormals` would be close to pointless since it's very easy to introduce the same issue elsewhere.
Maybe in the future.